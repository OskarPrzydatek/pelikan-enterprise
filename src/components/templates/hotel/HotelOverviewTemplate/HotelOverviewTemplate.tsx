import { OverviewListItem } from '~/components/molecules';
import { OverviewList, PageLayout } from '~/components/organisms';
import { IHotelItem, IOverviewTemplate } from '~/models';

interface IHotelOverviewTemplate extends IOverviewTemplate {
  data: IHotelItem[] | undefined;
}

export const HotelOverviewTemplate: React.FC<IHotelOverviewTemplate> = ({
  data,
  isLoading,
  error,
  page,
  onClickItem,
  onClickNavigate,
}: IHotelOverviewTemplate) => {
  const { isArray } = Array;

  return (
    <PageLayout error={error} isLoading={isLoading}>
      <OverviewList
        navigateLabel="Dodaj hotel"
        noItemsLabel="Brak hoteli w systemie"
        page={page}
        title="Hotele"
        onClickNavigate={onClickNavigate}
      >
        {isArray(data)
          ? data.map(({ id, name }) => (
              <OverviewListItem
                key={`${id}-${name}`}
                buttonDataTestID={`hotel-overview-list-item-${id}`}
                name={name}
                onClickNavigate={() => onClickItem(id)}
              />
            ))
          : null}
      </OverviewList>
    </PageLayout>
  );
};
