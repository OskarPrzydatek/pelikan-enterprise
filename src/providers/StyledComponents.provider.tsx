import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
