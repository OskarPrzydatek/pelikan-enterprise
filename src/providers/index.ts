export { StyledComponentsProvider } from './StyledComponents.provider';
