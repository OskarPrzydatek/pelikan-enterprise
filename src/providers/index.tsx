import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
