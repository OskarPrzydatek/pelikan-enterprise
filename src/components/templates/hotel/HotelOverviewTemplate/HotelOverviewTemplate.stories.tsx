import { Meta, StoryObj } from '@storybook/react';

import { StyledComponentsProvider } from '~/providers';

import { HotelOverviewTemplate } from './HotelOverviewTemplate';

const meta: Meta<typeof HotelOverviewTemplate> = {
  title: 'Templates/HotelOverviewTemplate',
  component: HotelOverviewTemplate,
};

export default meta;
type Story = StoryObj<typeof HotelOverviewTemplate>;

export const HotelOverviewTemplatePrimmary: Story = {
  render: () => (
    <StyledComponentsProvider>
      <HotelOverviewTemplate
        onClickNavigate={() => console.log('onClickNavigate')}
      />
    </StyledComponentsProvider>
  ),
};
