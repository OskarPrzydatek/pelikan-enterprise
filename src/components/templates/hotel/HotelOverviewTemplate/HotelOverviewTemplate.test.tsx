import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { HotelOverviewTemplate } from './HotelOverviewTemplate';

const mockOnClickItem = vi.fn();
const mockOnClickNavigate = vi.fn();

describe('HotelOverviewTemplate', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <HotelOverviewTemplate
          items={[
            { id: 1, name: 'hotel1' },
            { id: 2, name: 'hotel2' },
            { id: 3, name: 'hotel3' },
            { id: 4, name: 'hotel4' },
            { id: 5, name: 'hotel5' },
          ]}
          onClickItem={mockOnClickItem}
          onClickNavigate={mockOnClickNavigate}
        />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });
});
