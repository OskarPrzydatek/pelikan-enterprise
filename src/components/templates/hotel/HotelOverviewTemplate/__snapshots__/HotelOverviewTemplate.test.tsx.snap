// Vitest Snapshot v1

exports[`HotelOverviewTemplate > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iAEawV eSwvqh"
      >
        <div
          class="sc-jcMfQk kxHmzN"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Hotele
          </h2>
          <ul
            class="sc-cabOPr byBExJ"
          >
            <li
              data-testid="no-items-label"
            >
              <p
                class="sc-pyfCe kcVRAg"
              >
                Brak hoteli w systemie
              </p>
            </li>
          </ul>
          <div
            class="sc-iTFTee dlSrFH"
          >
            <button
              class="sc-jSUZER fQVznz"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
            <p
              class="sc-pyfCe kcVRAg"
              data-testid="pagination-counter"
            >
              1
               / X
            </p>
            <button
              class="sc-jSUZER hjlqfP"
            >
              <svg
                data-testid="arrow-icon"
                fill="none"
                viewBox="0 0 40 26"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                  fill="#013059"
                />
              </svg>
            </button>
          </div>
          <button
            class="sc-hLBbgP kQYyQS"
          >
            Dodaj hotel
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iAEawV eSwvqh"
    >
      <div
        class="sc-jcMfQk kxHmzN"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Hotele
        </h2>
        <ul
          class="sc-cabOPr byBExJ"
        >
          <li
            data-testid="no-items-label"
          >
            <p
              class="sc-pyfCe kcVRAg"
            >
              Brak hoteli w systemie
            </p>
          </li>
        </ul>
        <div
          class="sc-iTFTee dlSrFH"
        >
          <button
            class="sc-jSUZER fQVznz"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
          <p
            class="sc-pyfCe kcVRAg"
            data-testid="pagination-counter"
          >
            1
             / X
          </p>
          <button
            class="sc-jSUZER hjlqfP"
          >
            <svg
              data-testid="arrow-icon"
              fill="none"
              viewBox="0 0 40 26"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 13L39.75 0.00961876V25.9904L0 13Z"
                fill="#013059"
              />
            </svg>
          </button>
        </div>
        <button
          class="sc-hLBbgP kQYyQS"
        >
          Dodaj hotel
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
